src/lib/encoder/SimpleEncoder.h

```diff
@@ -16,18 +16,19 @@ class SimpleEncoder
 {
 public:
 	SimpleEncoder(unsigned ecc_bytes=40, unsigned bits_per_symbol=0, unsigned bits_per_color=0);
+	void set_encode_id(uint8_t encode_id); // [0-127] -- the high bit is ignored.
 
 	template <typename STREAM>
 	std::optional<cv::Mat> encode_next(STREAM& stream);
 
 	template <typename STREAM>
 	fountain_encoder_stream::ptr create_fountain_encoder(STREAM& stream, int compression_level=6);
 
-
 protected:
 	unsigned _eccBytes;
 	unsigned _bitsPerSymbol;
 	unsigned _bitsPerColor;
+	uint8_t _encodeId = 0;
```

Comment:
Ideally this would auto-increment, but it's not clear who should own the increment. The *decoder* is the part of the system that would like encode_id to be distinct (within its life cycle), but the decoder's life cycle is quite different from the encoder's!

A random encode_id (and increment from there) might be reasonable.